integration_tests/e2e/goal/addNote.cy.ts

```diff
@@ -8,7 +8,52 @@ context('Add a note', () => {
     cy.task('reset')
     cy.task('stubSignInAsUserWithEditAuthority')
     cy.task('stubAuthUser')
-    cy.task('getPrisonerById')
+    cy.task('getPrisonerById', 'G6115VJ')
+    cy.task('getPrisonerById', 'H4115SD')
+  })
+
+  it('should not be able to navigate directly to add note given Create Goal and Add Step has not been submitted', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    cy.signIn()
+
+    // When
+    cy.visit(`/plan/${prisonNumber}/goals/add-note`)
+
+    // Then
+    const createGoalPage = Page.verifyOnPage(CreateGoalPage)
+    createGoalPage.isForPrisoner(prisonNumber)
+  })
+
+  it('should not be able to arrive on add note page, then change the prison number in the URL', () => {
+    // Given
+    const prisonNumber = 'G6115VJ'
+    cy.signIn()
+    cy.visit(`/plan/${prisonNumber}/goals/create`)
+
+    let createGoalPage = Page.verifyOnPage(CreateGoalPage)
+    createGoalPage //
+      .setGoalTitle('Learn French')
+      .setGoalReviewDate(23, 12, 2024)
+      .submitPage()
+
+    const addStepPage = Page.verifyOnPage(AddStepPage)
+    addStepPage //
+      .setStepTitle('Book French course')
+      .setStepTargetDate(23, 12, 2024)
+      .submitPage()
+
+    const addNotePage = Page.verifyOnPage(AddNotePage)
+    addNotePage.isForPrisoner(prisonNumber)
+
+    const someOtherPrisonNumber = 'H4115SD'
+
+    // When
+    cy.visit(`/plan/${someOtherPrisonNumber}/goals/add-note`)
+
+    // Then
+    createGoalPage = Page.verifyOnPage(CreateGoalPage)
+    createGoalPage.isForPrisoner(someOtherPrisonNumber)
   })
 
```

Comment:
There is a scenario that this code does not yet address - that being where the user has arrived on page 1 (Describe a Goal); they have successfully submitted it and arrived on page 2 (Add a Step). But instead of submitting the Add A Step page, then change the URL in the address bar to `/add-note` (but for the same prisoner)
In this scenario we would arrive on the Add Note page. The CreateGoalForm object exists in the session for the correct prisoner 👍 , but there is no populated AddStepForm object in the session because the user never actually submitted page 2 (Add a Step)

I decided against protecting against this until we have our collection of AddStepForms in the session (rather than just one). Once that work is complete we can come back here and implement it